packages/formation/js/back-to-top.js

```diff
@@ -42,7 +42,7 @@ export function closure(button, buttonContainer, footer, buttonClasses) {
       hasHitBreakpoint = !hasHitBreakpoint;
     }
 
-    if (footerCheck() !== footerVisChanged) {
+    if (footer && footerCheck() !== footerVisChanged) {
```

Comment:
But it isn't clear why this is here.

Reply:
before this change we checked if `footer` existed before running either `breakpointCheck` or `footerCheck`. Since I removed the earlier check for `footer` I moved that check here, where it gets checked before the code that relies on it.

Reply:
Without this addition, we just run `footerCheck()`, which runs `isScrolledIntoView(footer)`. It seems like that is returning truthy which is a problem. The error I'm seeing is `Unable to get property getBoundingClientRect`. Rather than avoid calling the function, what do you think about modifying the helper to return false if the element isn't there? Optional chaining may work:

```js
/**
 * Check if an element is scrolled into view.
 * Returns false if the element is null
 */
function isScrolledIntoView(el) {
  const elemTop = el?.getBoundingClientRect().top;
  // Only partially || completely visible elements return true
  return elemTop >= 0 && elemTop <= window.innerHeight;
}
```

(This is untested)

Reply:
The transpiled version of this code is `el && el.getBoundingClientRect` which is functionally the same as `footer && footerCheck`, only checking for the existence of the element before even calling `footerCheck` avoids that unnecessary call entirely.

Reply:
Even if `footer && footerCheck()` stays in, I think it is better for us to return `false` when `footer` is `null` rather than throw an unexpected error.